### ISSUE TITLE
Improve diagnostic quality for less accessible base type argument

### ISSUE THREAD

**Issue by 333fred** [dotnet/roslyn#40977]
```cs
class A { }
public class B<T> { }
public class C : B<A> { }
```

Currently, the error is:
```
Severity Code    Description
Error    CS0060  Inconsistent accessibility: base class 'B<A>' is less accessible than class 'C'
```
This isn't great wording. It's the _type argument_ of the base class that's less accessible, but the wording implies that it's `B<T>` itself that's less accessible. The error should explicitly call out the type argument as the thing not accessible.

If someone from the community is interested in improving this, a change would need to be made around here: https://github.com/dotnet/roslyn/blob/master/src/Compilers/CSharp/Portable/Symbols/Source/SourceNamedTypeSymbol_Bases.cs#L316-L320, splitting this up into multiple checks.

**Comment by stbau04** [comment]
@333fred is this still of interest? Would have some time (and motivation)

**Comment by 333fred** [comment]
@stbau04 absolutely, feel free to take a crack at it.

**Comment by stbau04** [comment]
@333fred Is this error expected to only mention the _first_ type argument that is less accessible, or _all_ type arguments that are less accessible?
(if only the first one should be reported, [TypeSymbolExtensions.VisitType](https://github.com/dotnet/roslyn/blob/d271551a53301734aeb3f1c0f4f574c6af213527/src/Compilers/CSharp/Portable/Symbols/TypeSymbolExtensions.cs#L750) would already return the necessary symbol; otherwise i would adapt this method to get an option to find all matches)

(Edit: Asking before implementing anything, as VisitType is used quite often)

**Comment by 333fred** [comment]
All would be best, but even just first would be an improvement on the current behavior if all is too complicated to implement.
